components/mediation/tasks/org.wso2.micro.integrator.ntask.core/src/main/java/org/wso2/micro/integrator/ntask/core/internal/TasksDSComponent.java

```diff
@@ -141,21 +151,70 @@ protected void activate(ComponentContext ctx) {
                                                                                     clusterCoordinator, resolver);
                 // join cluster
                 clusterCoordinator.startCoordinator();
+                setSchedulerProperties();
                 coordinatedTaskScheduleManager.startTaskScheduler("");
             }
         } catch (Throwable e) {
             log.error("Error in initializing Tasks component: " + e.getMessage(), e);
         }
     }
 
+    private void setSchedulerProperties() {
+
+        Map<String, Object> configs = ConfigParser.getParsedConfigs();
+        Object exePeriod = configs.get(TASK_CONFIG + "." + RESOLVING_PERIOD);
+        if (exePeriod != null) {
+            int executionPeriod = Integer.parseInt(exePeriod.toString());
+            if (executionPeriod < 1) {
+                log.warn(
+                        RESOLVING_PERIOD + " should be greater than or equal to 1 seconds. The value " + executionPeriod
+                                + " will be ignored.");
+            } else {
+                CoordinatedTaskScheduleManager.setExecutionPeriod(executionPeriod);
+            }
+        }
+        Object resolvingFreq = configs.get(TASK_CONFIG + "." + RESOLVING_FREQUENCY);
+        if (resolvingFreq != null) {
+            int resolvingFrequency = Integer.parseInt(resolvingFreq.toString());
+            if (resolvingFrequency < 1) {
+                log.warn(RESOLVING_FREQUENCY + " should be greater than or equal to 1. The value " + resolvingFrequency
```

Comment:
In this case, the default will be used right? Shall we also mention that the default value of 5 will be used? same with the resolving period?

Reply:
Fixed